### ISSUE TITLE
Error while deploying AudioVideoPlayback on Azure Cloud Services

### ISSUE THREAD

**Issue by JonatanMedinilla** [microsoftgraph/microsoft-graph-comms-samples#218]
I'm trying to run the AudioVideoPlayback bot on Azure Cloud Services but the deployment is failing. In order to deploy it, I've followed the getting started steps of the sample:

- Create the certificate (a non-self-signed) for my domain "bot.jhondoe.com" as alias for "bot.cloudapp.net"

- Upload the certificate

- Configure the solution running the following script: 
`.\configure_cloud.ps1 -p {path to project} -dns bot.jhondoe.com -cn bot.jhondoe.com -thumb {thumbprint} -bid {botName} -aid {appId} -as {appSecret}`

- Generate the package

- Upload the package to the cloud service.

 
When the worker roles role instances are starting, they throw the following exception:
`Unhandled Exception: No such host is known at System.Net.Dns.GetAddrInfo(String name) at System.Net.Dns.InternalGetHostByName(String hostName, Boolean includeIPv6) at System.Net.Dns.GetHostEntry(String hostNameOrAddress) at Sample.AudioVideoPlaybackBot.WorkerRole.AzureConfiguration.GetInstancePublicIpAddress(String publicFqdn) in C:\repos\microsoft-graph-comms-samples\Samples\V1.0Samples\LocalMediaSamples\AudioVideoPlaybackBot\WorkerRole\AzureConfiguration.cs:line 462 at Sample.AudioVideoPlaybackBot.WorkerRole.AzureConfiguration.Initialize() in C:\repos\microsoft-graph-comms-samples\Samples\V1.0Samples\LocalMediaSamples\AudioVideoPlaybackBot\WorkerRole\AzureConfiguration.cs:line 313 at Sample.AudioVideoPlaybackBot.WorkerRole.WorkerRole.OnStart() in C:\repos\microsoft-graph-comms-samples\Samples\V1.0Samples\LocalMediaSamples\AudioVideoPlaybackBot\WorkerRole\WorkerRole.cs:line 92 at Microsoft.WindowsAzure.ServiceRuntime.RoleEnvironment.InitializeRoleInternal(RoleType roleTypeEnum) at Microsoft.WindowsAzure.ServiceRuntime.Implementation.Loader.RoleRuntimeBridge.<InitializeRole>b__0() at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state) at System.Threading.ThreadHelper.ThreadStart() '`

For some reason the worker role instance is not finding the Host Entry for "pip.0.bot.jhondoe.com" instance host name.  I'm guessing that the issue is related to the certificate or the solution's configuration but I don't know where to start.

**Comment by ssulzer** [comment]
@JonatanMedinilla 
Does the Cloud Service have public IP addresses (IL-PIPs) provisioned for each VM instance?
From the exception it seems like the bot startup code is not able to resolve the public IP address of the VM/WorkerRole instance.
https://docs.microsoft.com/en-us/previous-versions/azure/virtual-network/virtual-networks-instance-level-public-ip

**Comment by JonatanMedinilla** [comment]
@ssulzer Thanks fort the replay. I don't know if it's an error in the getting started of the samples or I misunderstood one of the steps; but in the step 3 of the  **Deploy** section, the guide says that the CNAME should be the same as our DNS. While analyzing the code, I found that the DNS is being used to generate the instance host name _pip.instanceNumber.dns_ (e.g.: pip.1.bot.jhondoe.com), and the only way that I got to get the Host Entry was using  _cloudapp.net_ as DNS (e.g.: pip.1.bot.cloudapp.net). But, in order to establish the communication with the media server, the CName must be under the domain of my  wildcard certificate (e.g.: bot.jhondoe.com). So, to solve the issue, I've configured the solution running the configure_cloud script with the following values.

`.\configure_cloud.ps1 -p {path to project} -dns bot.cloudapp.net -cn bot.jhondoe.com -thumb {thumbprint} -bid {botName} -aid {appId} -as {appSecret}`
